### PR TITLE
fix closure issues with async wasm compilation

### PR DESCRIPTION
**Pull request by kripken**
And add testing.

### CODE REVIEW

**Comment by dschuff** [comment]
What's the underlying issue here? Is use of object['field'] vs object.field a pattern we should follow elsewhere, or everywhere?

**Comment by kripken** [comment]
We need to do it whenever closure compiler could be confused. The main rules are that if we create the object, it's ok to use normal names, as closure will minify both their creation and use the same way. The risk is when the name is created somewhere else, like an event or object created in the browser.

**Comment by kripken** [comment]
(In both fixes here, the JS is receiving objects from the browser that closure knows nothing about.)

File: tests/test_browser.py [hunk 1]
```diff
@@ -3317,7 +3317,8 @@ def test_binaryen_native(self):
         ['-O2', '-s', 'EMTERPRETIFY=1'],
         ['-O2', '-s', 'ALLOW_MEMORY_GROWTH=1'],
         ['-O2', '-s', 'EMTERPRETIFY=1', '-s', 'ALLOW_MEMORY_GROWTH=1'],
-        ['-O2', '-s', 'OUTLINING_LIMIT=1000']
+        ['-O2', '-s', 'OUTLINING_LIMIT=1000'],
+        ['-O2', '--closure', '1'],
```
Comment (jgravelle-google) [inline tests/test_browser.py]: Yay trailing commas :)

**Comment by kripken** [comment]
Sure.

**Pull request merged**